plugins/upload-assets-to-s3/src/tasks/upload-assets-to-s3.ts

```diff
@@ -64,9 +65,16 @@ export default class UploadAssetsToS3 extends Task<typeof UploadAssetsToS3Schema
             CacheControl: options.cacheControl
           }
           currentBucket = params.Bucket
-          const data = await s3.upload(params).promise()
-          this.logger.info(`Uploaded ${styles.filepath(filepath)} to ${styles.URL(data.Location)}`)
+          const data = s3.upload(params).promise()
+          uploadPromises.push(data)
         }
+        await Promise.all(uploadPromises).then((resolvedUploadPromises) => {
```

Comment:
Interesting... we do want to throw and exit the process if any of them fail as this task is used in deployment (on CircleCI) and we don't want partial upload to equal success. We could use all `allSettled` though and throw once all the promises are returned with a collated list of the failures.

Reply:
either is probably fine, it's a case of whether we want to fail fast or not if something goes wrong. i guess we'd rather have the rest of the assets uploaded even if one upload fails rather than leaving the assets partially updated? in which case `allSettled` would be the choice.

Reply:
When I change the code to `Promise.allSettled` I get the error 
```
Property 'allSettled' does not exist on type 'PromiseConstructor'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2020' or later.ts(2550)
```
I actually think it's fine with `all` - updating it to `allSettled` makes the code slightly more difficult to read and doesn't add lots in terms of functionality. For the use case that I'm aware of where this task is used for deployment, the deployment will either pass or fail, and in the latter case will need to be fixed (in theory!) regardless of how many files successfully upload. 

Either way I have added a card to add support for ES2020 as this will not be the last time ES2020 features will be useful to us: https://financialtimes.atlassian.net/browse/CPP-1248